plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergMetadata.java

```diff
@@ -1565,6 +1565,24 @@ public void renameColumn(ConnectorSession session, ConnectorTableHandle tableHan
         }
     }
 
+    @Override
+    public void setColumnType(ConnectorSession session, ConnectorTableHandle tableHandle, ColumnHandle columnHandle, io.trino.spi.type.Type type)
+    {
+        IcebergTableHandle table = (IcebergTableHandle) tableHandle;
+        IcebergColumnHandle column = (IcebergColumnHandle) columnHandle;
+        verify(column.isBaseColumn(), "Cannot change nested field types");
+
+        Table icebergTable = catalog.loadTable(session, table.getSchemaTableName());
+        try {
+            icebergTable.updateSchema()
+                    .updateColumn(column.getName(), toIcebergType(type).asPrimitiveType())
```

Comment:
What about allowing `array(integer)` to become `array(bigint)` 
or `row(...)` to gain a new field?

Reply:
We can't do that unless fixing Iceberg if my understanding is correct. Changing nested field types (e.g. `row(x integer) -> row(x bigint)`) in row columns is possible without changing Iceberg.